src/components/collection-progress-group/helpers/parse-collection-progress.ts

```diff
@@ -155,4 +116,49 @@ function getNextTutorialCta({
 	}
 }
 
-export { parseCollectionProgress }
+/**
+ * Determine the "next tutorial" in a collection.
+ *
+ * For in-progress collections, the "next tutorial" is the first tutorial
+ * in the collection that does not have "complete" as its progress status.
+ *
+ * For completed collections, and for all unauthenticated cases,
+ * the "next tutorial" is the first tutorial in the collection.
+ */
+function getNextTutorial({
```

Comment:
Pulled out logic to compute `targetTutorial`, as suggested [here](https://github.com/hashicorp/dev-portal/pull/989#discussion_r975461445)